diff --git a/tracker/dmlc_tracker/local.py b/tracker/dmlc_tracker/local.py
index 6e4af12..09df15e 100644
--- a/tracker/dmlc_tracker/local.py
+++ b/tracker/dmlc_tracker/local.py
@@ -26,6 +26,13 @@ def exec_cmd(cmd, num_attempt, role, taskid, pass_env):
     num_retry = env.get('DMLC_NUM_ATTEMPT', num_attempt)
     num_trial = 0
 
+    cmd_str = ''
+    for k, v in env.items():
+        if str(k)[0:4] == 'DMLC':
+            strenv = str(k) + '=' + str(v);
+            cmd_str = cmd_str + ' ' + strenv
+    cmdline = cmdline + ' ' + cmd_str
+
     logging.debug('num of retry %d',num_retry)
 
     while True:
