diff --git a/src/allreduce_base.cc b/src/allreduce_base.cc
index 143db6e..a6daf20 100644
--- a/src/allreduce_base.cc
+++ b/src/allreduce_base.cc
@@ -486,12 +486,13 @@ AllreduceBase::TryAllreduceTree(void *sendrecvbuf_,
     // select must return
     watcher.Poll();
     // exception handling
-    for (int i = 0; i < nlink; ++i) {
+    //FIXME:workaround for Occlum
+    /*for (int i = 0; i < nlink; ++i) {
       // recive OOB message from some link
       if (watcher.CheckExcept(links[i].sock)) {
         return ReportError(&links[i], kGetExcept);
       }
-    }
+    }*/
     // read data from childs
     for (int i = 0; i < nlink; ++i) {
       if (i != parent_index && watcher.CheckRead(links[i].sock)) {
@@ -641,12 +642,13 @@ AllreduceBase::TryBroadcast(void *sendrecvbuf_, size_t total_size, int root) {
     // select
     watcher.Poll();
     // exception handling
-    for (int i = 0; i < nlink; ++i) {
+    //FIXME:workaround for Occlum
+    /*for (int i = 0; i < nlink; ++i) {
       // recive OOB message from some link
       if (watcher.CheckExcept(links[i].sock)) {
         return ReportError(&links[i], kGetExcept);
       }
-    }
+    }*/
     if (in_link == -2) {
       // probe in-link
       for (int i = 0; i < nlink; ++i) {
