From 64bafe1617c776fcbcb5e1587841396dc62abdc5 Mon Sep 17 00:00:00 2001
From: LI Qing <geding.lq@alibaba-inc.com>
Date: Mon, 13 Jan 2020 06:22:07 +0000
Subject: [PATCH] Modify Cargo.toml's dependency to use path instead of git url

---
 Cargo.toml | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/Cargo.toml b/Cargo.toml
index 096e9f3..bab974a 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -17,9 +17,9 @@ travis-ci = { repository = "serde-rs/json" }
 appveyor = { repository = "serde-rs/json" }
 
 [dependencies]
-serde = { git = "https://github.com/mesalock-linux/serde-sgx.git" }
-indexmap = { git = "https://github.com/mesalock-linux/indexmap-sgx.git", optional = true }
-itoa = { git = "https://github.com/mesalock-linux/itoa-sgx.git" }
+serde = { path = "../serde-sgx/serde" }
+#indexmap = { git = "https://github.com/mesalock-linux/indexmap-sgx.git", optional = true }
+itoa = { path = "../itoa-sgx" }
 ryu = "1.0"
 sgx_tstd = { rev = "v1.1.0", git = "https://github.com/apache/teaclave-sgx-sdk.git", optional = true }
 
@@ -49,7 +49,7 @@ default = ["mesalock_sgx"]
 # Use a different representation for the map type of serde_json::Value.
 # This allows data to be read into a Value and written back to a JSON string
 # while preserving the order of map keys in the input.
-preserve_order = ["indexmap"]
+#preserve_order = ["indexmap"]
 
 # Use an arbitrary precision number representation for serde_json::Number. This
 # allows JSON numbers of arbitrary size/precision to be read into a Number and
-- 
2.7.4

